adaptive/learner/learner1D.py

```diff
@@ -24,12 +24,22 @@
     partial_function_from_dataframe,
 )
 
+if TYPE_CHECKING:
+    import holoviews
+
 try:
     from typing import TypeAlias
 except ImportError:
     # Remove this when we drop support for Python 3.9
     from typing_extensions import TypeAlias
 
+try:
+    from typing import Literal
+except ImportError:
+    # Remove this when we drop support for Python 3.7
+    from typing_extensions import Literal
+
+
```

Comment:
I think we can drop 3.7, see https://numpy.org/neps/nep-0029-deprecation_policy.html

We didn't discuss this, but NEP 29 is a good source.